pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go

```diff
@@ -78,28 +78,22 @@ func NewBalancedAllocation(baArgs runtime.Object, h framework.Handle, fts featur
 		return nil, err
 	}
 
-	resToWeightMap := make(resourceToWeightMap)
-
-	for _, resource := range args.Resources {
-		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
-	}
-
 	return &BalancedAllocation{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
-			Name:                BalancedAllocationName,
-			scorer:              balancedResourceScorer,
-			useRequested:        true,
-			resourceToWeightMap: resToWeightMap,
+			Name:         BalancedAllocationName,
+			scorer:       balancedResourceScorer,
+			useRequested: true,
+			resources:    args.Resources,
 		},
 	}, nil
 }
 
-func balancedResourceScorer(requested, allocable resourceToValueMap) int64 {
+func balancedResourceScorer(requested, allocable []int64) int64 {
 	var resourceToFractions []float64
 	var totalFraction float64
-	for name, value := range requested {
-		fraction := float64(value) / float64(allocable[name])
+	for i, value := range requested {
+		fraction := float64(value) / float64(allocable[i])
```

Comment:
If we're for the performance objectives, use `requested[i]` instead.

Reply:
> If we're for the performance objectives, use `requested[i]` instead.

Got it.